omod/src/main/java/org/openmrs/module/fhir2/providers/LocationFhirResourceProvider.java

```diff
@@ -56,37 +58,13 @@ public Location getLocationById(@IdParam @NotNull IdType id) {
 	
 	@Search
 	@SuppressWarnings("unused")
-	public Bundle findLocationByName(@RequiredParam(name = Location.SP_NAME) StringParam name) {
-		return FhirUtils.convertSearchResultsToBundle(fhirLocationService.findLocationByName(name.getValue()));
-	}
-	
-	@Search
-	@SuppressWarnings("unused")
-	public Bundle findLocationByCity(@RequiredParam(name = Location.SP_ADDRESS_CITY) StringParam city) {
-		return FhirUtils.convertSearchResultsToBundle(fhirLocationService.findLocationsByCity(city.getValue()));
-	}
-	
-	@Search
-	@SuppressWarnings("unused")
-	public Bundle findLocationByCountry(@RequiredParam(name = Location.SP_ADDRESS_COUNTRY) StringParam country) {
-		return FhirUtils.convertSearchResultsToBundle(fhirLocationService.findLocationsByCountry(country.getValue()));
-	}
-	
-	@Search
-	@SuppressWarnings("unused")
-	public Bundle findLocationByPostalCode(@RequiredParam(name = Location.SP_ADDRESS_POSTALCODE) StringParam postalCode) {
-		return FhirUtils.convertSearchResultsToBundle(fhirLocationService.findLocationsByPostalCode(postalCode.getValue()));
-	}
-	
-	@Search
-	@SuppressWarnings("unused")
-	public Bundle findLocationByState(@RequiredParam(name = Location.SP_ADDRESS_STATE) StringParam state) {
-		return FhirUtils.convertSearchResultsToBundle(fhirLocationService.findLocationsByState(state.getValue()));
-	}
-	
-	@Search
-	@SuppressWarnings("unused")
-	public Bundle findLocationsByTag(@RequiredParam(name = "_tag") TokenParam tag) {
-		return FhirUtils.convertSearchResultsToBundle(fhirLocationService.findLocationsByTag(tag));
+	public Bundle searchLocations(@OptionalParam(name = Location.SP_NAME) StringParam name,
+	        @OptionalParam(name = Location.SP_ADDRESS_CITY) StringParam city,
+	        @OptionalParam(name = Location.SP_ADDRESS_COUNTRY) StringParam country,
+	        @OptionalParam(name = Location.SP_ADDRESS_POSTALCODE) StringParam postalCode,
+	        @OptionalParam(name = Location.SP_ADDRESS_STATE) StringParam state, @OptionalParam(name = "_tag") TokenParam tag,
```

Comment:
You can replace the literal `"_tag"` by `PARAM_TAG` present in `ca.uhn.fhir.rest.api.Constants`